packages/block-library/src/navigation/edit/index.js

```diff
@@ -256,16 +264,26 @@ function Navigation( {
 		setRef( fallbackNavigationMenus[ 0 ].id );
 	}, [ navigationMenus ] );
 
-	const navRef = useRef();
+	useEffect( async () => {
+		if (
+			! hasResolvedNavigationMenus ||
+			isConvertingClassicMenu ||
+			fallbackNavigationMenus?.length > 0 ||
+			classicMenus?.length !== 1
+		) {
+			return false;
+		}
 
-	const {
-		convert: convertClassicMenu,
-		status: classicMenuConversionStatus,
-		error: classicMenuConversionError,
-	} = useConvertClassicToBlockMenu( clientId );
+		// If there's non fallback navigation menus and
+		// only one classic menu then create a new navigation menu based on it.
+		await convertClassicMenu(
+			classicMenus[ 0 ].id,
+			classicMenus[ 0 ].name,
+			'publish'
+		);
```

Comment:
```suggestion
		await convertClassicMenu(
			classicMenus[ 0 ].id,
			classicMenus[ 0 ].name,
			'publish'
		).catch( () => {} );
```

We're not catching potential errors here. Ditto all other cases where we are `await`ing the result of the function.